straxen/plugins/nveto_recorder.py

```diff
@@ -53,54 +53,60 @@ def infer_dtype(self):
-                                self.config['coincidence_level_recorder_nv'],
-                                self.config['resolving_time_recorder_nv'])
-        # Always save the first and last resolving_time nanoseconds (e.g. 600 ns)  since we cannot guarantee the gap
-        # size to be larger. (We cannot use an OverlapingWindow plugin either since it requires disjoint objects.)
-        if len(intervals):
-            intervals_with_bounds = np.zeros((len(intervals) + 2, 2), dtype=np.int64)
-            intervals_with_bounds[1:-1, :] = intervals
-            intervals_with_bounds[0, :] = start, min(start + self.config['resolving_time_recorder_nv'], intervals[0, 0])
-            intervals_with_bounds[-1, :] = max(end - self.config['resolving_time_recorder_nv'], intervals[-1, 1]), end
-            del intervals
+        if self.config['coincidence_level_recorder_nv'] > 1:
+            intervals = coincidence(raw_records_nv,
+                                    self.config['coincidence_level_recorder_nv'],
+                                    self.config['resolving_time_recorder_nv'])
+            # Always save the first and last resolving_time nanoseconds (e.g. 600 ns)  since we cannot guarantee the gap
+            # size to be larger. (We cannot use an OverlapingWindow plugin either since it requires disjoint objects.)
+            if len(intervals):
+                intervals_with_bounds = np.zeros((len(intervals) + 2, 2), dtype=np.int64)
+                intervals_with_bounds[1:-1, :] = intervals
+                intervals_with_bounds[0, :] = start, min(start + self.config['resolving_time_recorder_nv'], intervals[0, 0])
+                intervals_with_bounds[-1, :] = max(end - self.config['resolving_time_recorder_nv'], intervals[-1, 1]), end
+                del intervals
+            else:
+                intervals_with_bounds = np.zeros((0, 2), dtype=np.int64)
+
+            neighbors = strax.record_links(raw_records_nv)
+            mask = pulse_in_interval(raw_records_nv, neighbors, *np.transpose(intervals_with_bounds))
+            rr, lone_records = straxen.mask_and_not(raw_records_nv, mask)
+
+            # Compute some properties of the lone_records:
+            # We compute only for lone_records baseline etc. since
+            # raw_records_nv will be deleted, otherwise we could not change
+            # the settings and reprocess the data in case of raw_records_nv
+            lr = strax.raw_to_records(lone_records)
+            del lone_records
+
+            lr = strax.sort_by_time(lr)
+            strax.zero_out_of_bounds(lr)
+            strax.baseline(lr,
+                           baseline_samples=self.config['nbaseline_samples_lone_records_nv'],
+                           flip=True)
+            strax.integrate(lr)
+            lrs, lr = compute_lone_records(lr, self.config['channel_map']['nveto'], self.config['n_lone_records_nv'])
+            lrs['time'] = start
+            lrs['endtime'] = end
         else:
```

Comment:
I would revert the order to use less indents and make the code much easier to read:
```python
 if self.config['coincidence_level_recorder_nv'] == 0:
        return {'raw_records_coin_nv': raw_records_nv,
                'lone_raw_records_nv': np.zeros(0, dtype=self.dtype['lone_raw_records_nv']),
                'lone_raw_record_statistics_nv': lrs = np.zeros(0, dtype=self.dtype['lone_raw_record_statistics_nv'])}
```